### ISSUE TITLE
Running locally, rather than live demo, the demo fails with errors

### ISSUE THREAD

**Issue by rburkey2005** [michaelfranzl/webAGC#1]
([This issue is a continuation of the discussion begun in the Virtual AGC repository.](https://github.com/virtualagc/virtualagc/pull/1144#issuecomment-849635334))

I understand that the shortcut "serve the parent directory via HTTP" is meaningful to web programmers, but less so to embedded programmers like myself.  In my case, not at all! :smile:  At any rate, I had to google it, and the solution I had settled on was precisely the ruby command you mention.  But I had ruby 2.5.1 rather than 3.0.1, so perhaps that's meaningful.

I just now tried your http-server idea.  (It requires a "-p 8000", by the way, because it defaulted to 8080.)  Unfortunately, with http-server I get precisely the same result as I did with ruby.  

I had installed node.js 14.17.0 this morning, which is the latest LTS release.  I notice that version 16 is also available; is it worth trying?

**Comment by michaelfranzl** [comment]
@rburkey2005 When I [suggested](https://github.com/virtualagc/virtualagc/pull/1144#issuecomment-849709528) to try `http-server`, I had actually tried it, and it had worked. But I must have fooled myself somehow (browser cache or wrong browser tab), because there is a long-open [issue](https://github.com/http-party/http-server/issues/480) with `http-server` reporting that it serves WebAssembly files wrongly. And now that I try it again, it is indeed broken in that regard. What can I say... more than a decade of web development experience, and I'm still repeatedly surprised! :smiley: Apologies for the hassle...

The background story is that software ecosystems are just waking up to the existence of WebAssembly, and their MIME types need some catching up.

In any event...

Please try one of the following tested (really, now! :wink: ) options:

Using Node.js. I tested this in version v14.17.0 and v16.2.0.

```sh
npm install --global light-server # installs version 2.9.1
cd path/to/webAGC
light-server --no-reload -p 8000 -s .
# Browse to http://localhost:8000/demo/
```

Using Ruby 3.0.1. I just confirmed that older versions of Ruby do _not_ set the WebAssembly content type correctly. Newer versions should work though.

```sh
# Prerequisite: Install Ruby 3.0.1 (via package manager, manual download, compilation from sources, rbenv, nvm, etc.)
gem install webrick
cd path/to/webAGC
ruby -r un.rb -e httpd . -p 8000
# Browse to http://localhost:8000/demo/
```

Please report back if a method worked for you. I'll then update the README.

**Comment by rburkey2005** [comment]
Regarding light-server ... the simulation seems to run, and the erasable memory area updates.  Yay!

Regarding ruby 2.5.1, I think you had mentioned "`gem install webrick`" only in connection with ruby 3.0.1.  But if I install webrick and use ruby 2.5.1, the simulation _seems_ to work okay.

Regarding ruby 3.0.1 (built from source) plus webrick, it seems to work too.

By the way, I've noticed that when I run Validation, which is what I normally have been doing, the DSKY sometimes misses the first PRO key, and requires a 2nd PRO to get started.  Once I'm _pretty_ sure that it never accepted a PRO key at all, and I ended up having to use the "DSKY key input".  I wonder if there's some sequencing issue at startup with "load program into fixed memory" vs Pause vs Reset vs Run that makes detection of at least the initial DSKY keypress intermittent?  I know that's probably too vague to be meaningful, sorry!  I suppose it may be some caching problem, since kept changing the server all the time.

Anyway, I'm happy, so I'm going to close this issue.  But if you have any more comments, I'd like to hear them.

**Comment by michaelfranzl** [comment]
@rburkey2005 You are right, `gem install webrick` will upgrade WEBrick to the newest version, and it will serve the `.wasm` file correctly, even with an older Ruby 2.5.3.

WEBrick version 1.4.2, which ships with Ruby 2.5.3, is too old. It got wasm support here: https://github.com/ruby/webrick/pull/17 Version 1.7.0 definitely works. I'll update the README file accordingly.

About the PRO key: I also found some irregularities (on my mobile phone it works for Luminary099, but doesn't for Validation, while on my desktop it always works). I'll investigate this as a separate issue. Thanks for reporting it!
